### PR TITLE
Add link to Catalyst device page

### PR DESCRIPTION
**Pull request by josh146**
**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-70009]

### CODE REVIEW

File: doc/introduction/compiling_workflows.rst [hunk 1]
```diff
@@ -21,9 +21,10 @@ Installing compilers
 --------------------
 
 Currently, Catalyst must be installed separately, and only supports the JAX
-interface and select devices such as ``lightning.qubit``,
-``lightning.kokkos``, ``braket.local.qubit`` and ``braket.aws.qubit``. It
-does **not** support ``default.qubit``.
+interface and select devices. Supported backend devices for Catalyst include
+``lightning.qubit``, ``lightning.kokkos``, ``lightning.gpu``, and ``braket.aws.qubit``,
+but **not** ``default.qubit``.
+For a full of supported devices, please see :doc:`catalyst:dev/devices`.
```
Comment (DSGuala) [inline doc/introduction/compiling_workflows.rst:27]:
```suggestion
For a full list of supported devices, please see :doc:`catalyst:dev/devices`.
```

File: pennylane/compiler/__init__.py [hunk 2]
```diff
@@ -102,12 +102,11 @@
...
-    ``lightning.kokkos``, ``braket.local.qubit``, and ``braket.aws.qubit``
-    devices. It does not support ``default.qubit``.
-
-    See the :doc:`Catalyst documentation <catalyst:index>` for more details on supported
-    devices, operations, and measurements.
+    Supported backend devices for Catalyst include
+    ``lightning.qubit``, ``lightning.kokkos``, ``lightning.gpu``, and ``braket.aws.qubit``,
+    but **not** ``default.qubit``.
+    
+    For a full of supported devices, please see :doc:`catalyst:dev/devices`.
```
Comment (DSGuala) [inline pennylane/compiler/__init__.py:109]:
```suggestion
    For a full list of supported devices, please see :doc:`catalyst:dev/devices`.
```

File: pennylane/compiler/qjit_api.py [hunk 3]
```diff
@@ -46,12 +46,12 @@ def qjit(fn=None, *args, compiler="catalyst", **kwargs):  # pylint:disable=keywo
...
-        devices. It does not support ``default.qubit``.
-
-        Please see the :doc:`Catalyst documentation <catalyst:index>` for more details on
-        supported devices, operations, and measurements.
+        Catalyst only support the JAX interface, and selected devices.
+        Supported backend devices for Catalyst include
+        ``lightning.qubit``, ``lightning.kokkos``, ``lightning.gpu``, and ``braket.aws.qubit``,
+        but **not** ``default.qubit``.
+        
+        For a full of supported devices, please see :doc:`catalyst:dev/devices`.
```
Comment (DSGuala) [inline pennylane/compiler/qjit_api.py:54]:
```suggestion
        For a full list of supported devices, please see :doc:`catalyst:dev/devices`.
```

File: pennylane/compiler/qjit_api.py [hunk 4]
```diff
@@ -46,12 +46,12 @@ def qjit(fn=None, *args, compiler="catalyst", **kwargs):  # pylint:disable=keywo
 
     .. note::
 
-        Catalyst supports compiling QNodes that use ``lightning.qubit``,
-        ``lightning.kokkos``, ``braket.local.qubit``, and ``braket.aws.qubit``
-        devices. It does not support ``default.qubit``.
-
-        Please see the :doc:`Catalyst documentation <catalyst:index>` for more details on
-        supported devices, operations, and measurements.
+        Catalyst only support the JAX interface, and selected devices.
```
Comment (DSGuala) [inline pennylane/compiler/qjit_api.py:49]:
```suggestion
        Catalyst only supports the JAX interface and selected devices.
```

**Review by DSGuala — COMMENTED** [review]
Looks good, I left some small copy update suggestions.

**Comment by josh146** [comment]
Nice catches @DSGuala, I made the changes too quickly 😓

**Review by DSGuala — APPROVED** [review]
Great! Looks ready to go :rocket:

**Pull request merged**